src/AlgebraicGeometry/Schemes/CoveredSchemes/Objects/Properties.jl

```diff
@@ -32,16 +32,35 @@ Return the boolean value whether a covered scheme `X` is smooth.
 is_smooth(X::AbsCoveredScheme) = is_smooth(underlying_scheme(X))
 
 @attr function is_smooth(X::CoveredScheme)
-  if !isdefined(X, :coverings) 
+  if !isdefined(X, :coverings)
     return true
   end
   # TODO: Can this be optimized? We only need to check the rank of
-  # the jacobian matrices where we haven't checked in another chart before. 
+  # the jacobian matrices where we haven't checked in another chart before.
   # This is a tradeoff between cost of Jacobian criterion and cost of
   # optimizing the use of the covering
   return all(is_smooth, affine_charts(X))
 end
 
+# Use decomposition_info
+function _jacobian_criterion(X::CoveredScheme{<:Field})
+  if !isdefined(X, :coverings)
+    return true
+  end
```

Comment:
what exactly do we do here? If we want to check that the scheme is empty,
then we better do `is_empty(X)`?

Reply:
We actually do need to check the existence of a covering, not only the emptiness of the scheme. We need the covering for the rest of the algorithm. Judging from the definition of `is_empty`, it seems that it is possible to have both empty covered schemes with a covering and empty covered schemes without a covering.